python/langsmith/client.py

```diff
@@ -151,11 +151,19 @@ def _default_retry_config() -> Retry:
     return Retry(**retry_params)  # type: ignore
 
 
-def _serialize_json(obj: Any) -> Any:
+_PRIMITIVE_TYPES = (str, int, float, bool)
+_MAX_DEPTH = 3
+
+
+def _serialize_json(obj: Any, depth: int = 0) -> Any:
```

Comment:
do we need to handle lists or dicts or tuples here?

Reply:
Updated